ksql-engine/src/test/java/io/confluent/ksql/util/ExecutorUtilTest.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package io.confluent.ksql.util;
+
+import org.apache.kafka.common.errors.RetriableException;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+
+public class ExecutorUtilTest {
+
+  @Rule
+  public final ExpectedException expectedException = ExpectedException.none();
+
+  @Test
+  public void shouldRetryAndEventuallyThrowIfNeverSucceeds() throws Exception {
+    expectedException.expect(ExecutionException.class);
+    expectedException.expectMessage("I will never succeed");
+
+    ExecutorUtil.executeWithRetries(() -> {
+          final CompletableFuture<Void> f = new CompletableFuture<>();
```

Comment:
Executor no longer uses Futures, so you can drop them from the test.

This can just be `throw new TestRetriableException(blah)` now...